components/Navbar/index.js

```diff
@@ -1,72 +1,98 @@
-import React from 'react';
-import { Hamburger, NavBarToggle, MainNav, SubNav, TableHeader, Block, Logo } from '@/styles';
+import React, { useState, useEffect, useRef } from 'react';
+import { Hamburger, NavBarToggle, MainNav, SubNav, TableHeader, Logo, HeaderImage, NavItem, Spacer } from '@/styles';
 import Image from 'next/image';
 import Link from 'next/link';
-import { useState, useEffect } from 'react';
+import { set } from 'mongoose';
 
-
-const Navbar = (props) => {
+const Navbar = () => {
     const [displayNav, setDisplayNav] = useState(false);
+    const [showCreateProfile, setShowCreateProfile] = useState(false);
+    const [showLists, setShowLists] = useState(false);
+    const dropdownRef = useRef(null);
 
     useEffect(() => {
         const handleResize = () => {
-            if (window.innerWidth <= 768) {
-                setDisplayNav(false);
-} else {
-                setDisplayNav(true);
+            setDisplayNav(window.innerWidth > 768);
+        };
+
+        const handleClickOutside = (event) => {
+            if (dropdownRef.current && !dropdownRef.current.contains(event.target)) {
+                setShowCreateProfile(false); 
+                setShowLists(false);
             }
-        }
+        };
 
-        window.addEventListener('resize', handleResize);
 
+        handleResize(); 
         return () => {
             window.removeEventListener('resize', handleResize);
-        }
+            document.removeEventListener('mousedown', handleClickOutside);
+        };
     }, []);
 
-    const toggleNavBar = () => {
-        setDisplayNav(!displayNav);
-    };
+    const toggleNavBar = () => setDisplayNav(!displayNav);
 
+    const toggleCreateProfile = () => setShowCreateProfile(!showCreateProfile);
+    const toggleLists = () => setShowLists(!showLists);
     return (
         <>
-<TableHeader>
-            <Logo>
-            <Link href="/">
-                <Image src="/logo.png" alt="logo" width={100} height={100} priority={true} />
-            </Link>
-            </Logo>
-            <Hamburger>
-                <NavBarToggle>
-                    <Image onClick={toggleNavBar} src="/hamburger.svg" alt="hamburger" width={30} height={30} />
-                </NavBarToggle>
-            </Hamburger>
+            <TableHeader>
+                <Hamburger>
+                    <NavBarToggle onClick={toggleNavBar}>
+                        <Image src="/hamburger.svg" alt="hamburger" width={30} height={30} />
+                    </NavBarToggle>
+                </Hamburger>
+                <MainNav style={{ display: displayNav ? 'flex' : 'none' }}>
```

Comment:
please avoid using inline style. Solve this instead by working with a prop and handling it in your styled component